src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -167,10 +167,25 @@ def _process_message(wrapper: Dict) -> None:
             try:
                 check_in = MonitorCheckIn.objects.select_for_update().get(
                     guid=params["check_in_id"],
-                    project_id=project_id,
-                    monitor=monitor,
                 )
 
+                if (
+                    check_in.project_id != project_id
+                    or check_in.monitor_id != monitor.id
+                    or check_in.monitor_environment_id != monitor_environment.id
+                ):
+                    metrics.incr(
+                        "monitors.checkin.result",
+                        tags={"source": "consumer", "status": "guid_mismatch"},
+                    )
+                    logger.debug(
```

Comment:
Worth making info in case you want to see it in gcp logs?

Reply:
we really just want the metric and then any spikes there will lead to an SDK investigation